src/pages/Login.tsx

```diff
@@ -0,0 +1,324 @@
+import React, { useEffect, useMemo } from 'react';
+import {
+  Box,
+  Button,
+  FormControl,
+  IconButton,
+  InputAdornment,
+  InputLabel,
+  OutlinedInput,
+  fabClasses,
+} from '@mui/material';
+import Image from 'next/image';
+import appLogo2 from '../../public/appLogo.png';
+import {
+  CloseFullscreen,
+  Visibility,
+  VisibilityOff,
+} from '@mui/icons-material';
+// import { useLocation } from 'react-router-dom';
+import { useRouter } from 'next/router';
+import CloseIcon from '@mui/icons-material/Close';
+import { useState } from 'react';
+// import '../App.css';
+import { login } from '../services/LoginService';
+import { useTranslation } from 'react-i18next';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import { useTheme } from '@mui/material/styles';
+import MenuItem from '@mui/material/MenuItem';
+import config from '../../config.json';
+// import { getUserId } from '../services/ProfileService';
+// import Loader from '../components/Loader.tsx';
+import Snackbar, { SnackbarOrigin } from '@mui/material/Snackbar';
+import Link from '@mui/material/Link';
+import Checkbox from '@mui/material/Checkbox';
+
+interface State extends SnackbarOrigin {
+  openModal: boolean;
+}
+
+const LoginPage = () => {
+  const { t, i18n } = useTranslation();
+  const [showPassword, setShowPassword] = useState(false);
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [showToastMessage, setShowToastMessage] = useState(false);
+  const [usernameError, setUsernameError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  //   const [selectedLanguage, setSelectedLanguage] = useState(
+  //     localStorage.getItem('preferredLanguage') || 'en'
+  //   );
+  const [selectedLanguage, setSelectedLanguage] = useState('en');
+  const [language, setLanguage] = useState(selectedLanguage);
+  const router = useRouter();
+  //   const location = useLocation();
+  const theme = useTheme<any>();
+  const [state, setState] = React.useState<State>({
+    openModal: false,
+    vertical: 'top',
```

Comment:
@suvarnakale move this to helper file or keep it in one object and reuse across file